src/vs/editor/common/controller/cursor.ts

```diff
@@ -1092,7 +1093,8 @@ export class Cursor extends EventEmitter {
 				postOperationRunnable: null,
 				shouldPushStackElementBefore: false,
 				shouldPushStackElementAfter: false,
-				requestScrollDeltaLines: 0
+				requestScrollDeltaLines: 0,
+				eventData: ctx.eventData
```

Comment:
Why? Please decode the arguments before invoking for each cursor. If there are 1000 cursors it makes no sense to pass the data to each cursor to have it parsed and decoded 1000 times.